src/contrib/cluster/Akka.Cluster.Sharding/ShardedDaemonProcess.cs

```diff
@@ -106,28 +151,35 @@ public class ShardedDaemonProcess : IExtension
         /// <param name="name">TBD</param>
         /// <param name="numberOfInstances">TBD</param>
         /// <param name="propsFactory">Given a unique id of `0` until `numberOfInstance` create an entity actor.</param>
-        public void Init(string name, int numberOfInstances, Func<int, Props> propsFactory) => 
+        /// <returns>A reference to a router actor that will distribute all messages evenly across the workers
+        /// using round-robin message routing. <c>null</c> if the ShardedDaemonProcess is misconfigured.</returns>
+        public IActorRef? Init(string name, int numberOfInstances, Func<int, Props> propsFactory) =>
```

Comment:
Modified all of these methods to return an `IActorRef?` - will return `null` if we're the wrong role type to be hosting a worker and we'll also now log a `WARNING` accordingly if that happens.